core/store/src/trie/split_state.rs

```diff
@@ -0,0 +1,372 @@
+use crate::trie::iterator::TrieItem;
+use crate::{get_delayed_receipt_indices, set, ShardTries, StoreUpdate, Trie, TrieUpdate};
+use borsh::BorshDeserialize;
+use near_primitives::account::id::AccountId;
+use near_primitives::errors::StorageError;
+use near_primitives::receipt::Receipt;
+use near_primitives::shard_layout::ShardUId;
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::{StateChangeCause, StateRoot};
+use std::collections::HashMap;
+use std::str::from_utf8;
+
+impl Trie {
+    /// Computes the set of trie items (nodes with keys and values) for a state part.
+    ///
+    /// # Panics
+    /// storage must be a TrieCachingStorage
+    /// part_id must be in [0..num_parts)
+    ///
+    /// # Errors
+    /// StorageError if the storage is corrupted
+    pub fn get_trie_items_for_part(
+        &self,
+        part_id: u64,
+        num_parts: u64,
```

Comment:
We might consider defining 

```rust
pub struct PartId {
  idx: u64,
  total: u64,
}
```

Reply:
Yeah that's a good suggestion. `PartId` is used in multiple other places too, so I would think it's better to do the refactoring in another PR. I'll create an issue for this.

Reply:
https://github.com/near/nearcore/issues/4806